sklearn/cluster/_agglomerative.py

```diff
@@ -915,15 +901,12 @@ class AgglomerativeClustering(ClusterMixin, BaseEstimator):
 
     _parameter_constraints: dict = {
         "n_clusters": [Interval(Integral, 1, None, closed="left"), None],
-        "affinity": [
-            Hidden(StrOptions({"deprecated"})),
-            StrOptions(set(_VALID_METRICS) | {"precomputed"}),
-            callable,
-        ],
         "metric": [
             StrOptions(set(_VALID_METRICS) | {"precomputed"}),
             callable,
-            None,
+            # equivalent to "euclidean" used during the deprecation of `affinity`
+            # parameter
+            Hidden(None),
```

Comment:
Should we deprecate None now ?

Reply:
I was unsure but it should not be disturbing actually.